### ISSUE TITLE
Using side-by-side with Firefox Sync

### ISSUE THREAD

**Issue by stormgrass** [marcelklehr/floccus#47]
Hello! I have a problem with using floccus on more than one browsers. This is what happens:
1. Set up floccus on first browser, bookmarks sync to Nextcloud.
2. Set up floccus on second browser. Another bookmarks folder is created, which is then synced with Nextcloud and the other browser.
3. Bookmarks syncing from second browser now overwrite bookmarks synced from first browser.

The result is an inconsistency within my Nextcloud bookmarks, resulting in different bookmarks depending on which browser has just synced to Nextcloud.

Is there a way to dedicate just one bookmarks folder and have floccus remember this across devices? That would be perfect.

Best,
Richard

**Comment by marcelklehr** [comment]
Hey! :)

Thanks for reaching out!

> The result is an inconsistency within my Nextcloud bookmarks, resulting in different bookmarks depending on which browser has just synced to Nextcloud.
> Is there a way to dedicate just one bookmarks folder and have floccus remember this across devices? That would be perfect.

I'm not sure I understand the problem. First things first though: Which version of floccus are you using? Then, can you describe in more detail how this inconsistency manifests and/or how to reproduce this?

i.e. I assume, you set up floccus in browser A, added some bookmarks into the floccus folder, set up floccus in browser B, and synced. Now all bookmarks should be in browser B's floccus folder, too, as well as in nextcloud. Where do you experience problems? When you set up floccus in browser B? When you add bookmarks in either browser?

**Comment by stormgrass** [comment]
Hi! Thanks for your response! 

I'm using Floccus 1.3.3 on FF Developer Edition 58.0b16.

Ok, so it went like this: I installed it on browser A, which created my Nextcloud bookmarks folder in the browser. I then set it up on browser B, which created _another_ Nextcloud bookmarks folder. I now had two separate folders. Deleting one of them resulted in syncing to Nextcloud not working anymore. Keeping both resulted in my Nextcloud bookmarks updating to whatever was inside the folder in the browser I was currently using.

So basically whenever I used one browser that held, say, five bookmarks, I'd have those five on Nextcloud. If I started using the other browser, which, let's say held ten different bookmarks, my Nextcloud would now hold those ten bookmarks, as opposed to the five ones it held previously.

So I think the problem is that with different bookmarks folders created when setting up the extension, only one folder is synced.

I hope that makes things a bit clearer :)

Best,
Richard

**Comment by marcelklehr** [comment]
Ok, it's getting clearer to me. The fact that the contents of the two folders don't converge is a problem. I'm not sure why it's a surprise to have a folder in both browsers, though. Where should floccus in browser B display the bookmarks to you, if not in a folder like browser A floccus does? Unless we are talking about the same browser installation and different profiles which share some amount of things?

**Comment by marcelklehr** [comment]
Also, after deleting a floccus folder, it should re-create and populate it the next time it syncs... :/

**Comment by stormgrass** [comment]
Ah, wait a second. I'm an idiot. 

I just realized that I've set up Firefox to sync all my bookmarks. Of course when I then set up floccus on one browser, that one folder will be synchronized to my other browser, where I'm guessing it interferes with my the setup of floccus there.

So I guess the only way to not have this behaviour would be to turn off bookmarks syncing in Firefox?

**Comment by marcelklehr** [comment]
Ah, that makes sense. I'm not sure why you'd want to have both. But it might work, if you only install floccus in one browser. (Which should be enough, given that you already sync via firefox)

**Comment by stormgrass** [comment]
True. But then I'd always have to wait until I'd be using the other browser with floccus installed in order for it to properly sync to the Nextcloud bookmarks app.

But yeah, that's for me to figure out. Thanks for taking the time to help untangle this for me.

**Comment by marcelklehr** [comment]
Ok, I get it. I'm not sure exactly what the interference is about, though...

In order to aid in debugging, you could provide client-side logs.

#### Firefox
* Go to `about:debugging`
* click on `debug` button next to floccus entry
* go to floccus options and make sure the account of your choice is connected
* trigger a sync by clicking on `force Sync` for the account of your choice.
* go back to the inspector window and copy the logs after waiting approx. 2mins
* paste the logs *somewhere else* removing any sensbile info and link to the site here (e.g. use https://pastebin.com)

#### Chrome
* Go to `chrome://extensions`
* enable `Developer mode`
* click on `dist/html/background.html` button in floccus' entry next to "Inspect views: "
* go to floccus options and make sure the account of your choice is connected
* trigger a sync by clicking on `force Sync` for the account of your choice
* go back to the inspector window and copy the logs after waiting approx. 2mins
* paste the logs *somewhere else* removing any sensbile info and link to the site here (e.g. use https://pastebin.com)

**Comment by sunjam** [comment]
Hi, I’ve started on a conversation with the Mozilla Sync team on how to improve sync support for selfhosting beyond what is currently available in 1.5.  [Please share your thoughts on the Nextcloud Help forum here](https://help.nextcloud.com/t/bring-firefox-sync-back/103) or [on Reddit here](https://support.mozilla.org/en-US/questions/1207473)

Please use the links provided to avoid hijacking this issue on Github.  Thanks!

**Comment by e-alfred** [comment]
@sunjam The Mozilla answer is basically useless. Firefox Sync 1.5 is a mess to set up, especially if completely selfhosted with storage + accounts server. They just want the data of their users, just as much as telemetry and other "diagnostic" information from their users.

The above problem also appears if another sync service for bookmarks is used (Chrome, Opera, Vivaldi).

**Comment by sunjam** [comment]
@e-alfred do you have any specific suggestions for how Mozilla could assist
the selfhosting community?

**Comment by e-alfred** [comment]
Sadly it is not as easy to set up a Sync server anymore as in the past with Sync 1.1. Back then, a few PHP files on a LAMP server were enough to make it run.

To make it a bit easier, they could provide packages (Debian/Red Hat/Centos...) in repositories or a Snap package for easy installation (and updates, which are a major problem in the current situation) for example. Mozilla already provides a Firefox package as a snap, so a Firefox Sync server (Accounts+Data) would be great too.

Another point would be to make it easier in Firefox directly to configure a custom Sync server. Currently, one needs to do some changes in about:config to run Firefox with your own Sync server.

**Comment by marcelklehr** [comment]
Thanks for the explanations @e-alfred!

It's confirmed that Firefox sync breaks floccus, so I'll close this as a won't fix. I'll also add recommendations to the README not to use other native syncing services.
